client/ipa-join.c

```diff
@@ -612,161 +632,417 @@ join_krb5(const char *ipaserver, char *hostname, char **hostdn, const char **pri
     return rval;
 }
 
+#else
```

Comment:
I assume at some point we're going to completely remove the xmlrpc-c support so it isn't a huge deal but it might be nice to have a comment here that it will use JSON (there is only one #define, XMLRPC or something else).